src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -176,10 +176,15 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
                 }
             }
 
-            final boolean useNewLikelihoods = newPLs != null;
-            final GenotypeBuilder gb = useNewLikelihoods ? new GenotypeBuilder(g).PL(newPLs).log10PError(newLog10GQ) : new GenotypeBuilder(g).noPL().noGQ();
-            GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePair = GenotypeLikelihoods.getAllelePair(MathUtil.indexOfMin(newPLs));
-            gb.alleles(Stream.of(allelePair.alleleIndex1, allelePair.alleleIndex2).map(allelesToKeep::get).collect(Collectors.toList()));
+            final GenotypeBuilder gb;
+            if (newPLs == null) {
+                gb = new GenotypeBuilder(g).noPL().noGQ().alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+            } else {
+                gb = new GenotypeBuilder(g).PL(newPLs).log10PError(newLog10GQ);
+                GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePair = GenotypeLikelihoods.getAllelePair(MathUtil.indexOfMin(newPLs));
```

Comment:
You don't need to get the allele pair, then the alleles list.  Instead, use `GenotypeLikelihoods.getAlleles(MathUtil.indexOfMin(newPLs), 2)` (which gets the intermediate allele pair behind the scenes).  The `2` is the ploidy.

Reply:
ah, thanks.